test/test_share_plus.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/services.dart';
+import 'package:share_plus/share_plus.dart';
+
+class MockSharePlus {
+  MockSharePlus();
+
+  /// The mock [ShareResult.raw] that `shareWithResult` should give.
+  String resultString = 'some-success-response';
+
+  /// The last string that `shareWithResult` was called with.
+  String? sharedString;
+
+  Future<Object?> handleMethodCall(MethodCall methodCall) async {
+    switch (methodCall.method) {
+      case 'shareWithResult':
+        // The method channel doesn't preserve Map<String, dynamic> as
+        // `arguments`'s type; logging runtimeType gives _Map<Object?, Object?>.
+        final arguments = methodCall.arguments as Map;
+        sharedString = arguments['text'] as String;
+        return resultString;
+      case 'shareUri':
+      case 'share':
+      case 'shareFiles':
+      case 'shareFilesWithResult':
+        throw UnimplementedError();
```

Comment:
Mmm, probably could replace these individual cases with a `default` case. When I added these individually, I think I assumed that method calls unrelated to `share_plus` might pass through here—like calls go through `SystemChannels.platform` whether or not they're about the clipboard, as we found with our `MockClipboard`.

Reply:
> When I added these individually, I think I assumed that method calls unrelated to `share_plus` might pass through here

(And they won't, because we're setting up this mock method-call handler on the plugin's own method channel that it creates.)

Reply:
Cool, that sounds good.

The question I was going to ask after reading this piece of code was what you referred to to identify this list of method names. But that's made moot if the list is removed.